src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -406,7 +406,7 @@ impl<'a, 'tcx> Decodable<DecodeContext<'a, 'tcx>> for ExpnId {
         let expn_cnum = Cell::new(None);
         let get_ctxt = |cnum| {
             expn_cnum.set(Some(cnum));
-            if cnum == LOCAL_CRATE {
+            if cnum == local_cdata.cnum {
```

Comment:
```suggestion
        let cstore = decoder.cdata().cstore;
        let get_ctxt = |cnum| {
            expn_cnum.set(Some(cnum));
            &cstore.get_crate_data(cnum).cdata.hygiene_context
```
`get_crate_data` is very cheap, so I'd just always use it for simplicity.